src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -656,4 +702,41 @@ public static void exitGameIfFinished() {
     });
   }
 
+  /**
+   * todo, replace with something better
+   * Get the chat for the game, or null if there is no chat.
+   */
+  public static Chat getChat() {
+    final ISetupPanel model = setupPanelModel.getPanel();
+    if (model instanceof ServerSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else if (model instanceof ClientSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else {
+      return null;
+    }
+  }
+
+  public static boolean hasChat() {
+    return getChat() != null;
+  }
+
+  /**
+   * After the game has been left, call this.
+   */
+  public static void clientLeftGame() {
+    if (!SwingUtilities.isEventDispatchThread()) {
+      SwingAction.invokeAndWait(GameRunner::clientLeftGame);
+      return;
+    }
+    // having an oddball issue with the zip stream being closed while parsing to load default game. might be caused by
+    // closing of stream while unloading map resources.
+    ThreadUtil.sleep(100);
+    setupPanelModel.showSelectType();
+    showMainFrame();
+  }
+
+  public static void quitGame() {
+    mainFrame.dispatchEvent(new WindowEvent(mainFrame, WindowEvent.WINDOW_CLOSING));
```

Comment:
Might want to guard against this running off the EDT as you did in `clientLeftGame()`.  Currently, the only caller (in `MainPanel`) is already running on the EDT, but you never know what someone will do in the future. :smile: